src/collector/WebPageLoadTimesCollector.js

```diff
@@ -0,0 +1,116 @@
+export default class WebPageLoadTimesCollector {
+
+  constructor() {
+    if (typeof window.performance.getEntriesByType !== 'undefined') {
+      this.currentPerformanceAPI = window.performance.getEntriesByType('navigation')[0];
+    }
+    this.oldPerformanceAPI = window.performance;
+  }
+
+  /**
+     * @param {Event} event
+     * @returns {Promise<*>|Event}
+     */
+  async prepare (event) {
+    event.eventDuration2 = WebPageLoadTimesCollector.getDOMInteractive(this.currentPerformanceAPI, this.oldPerformanceAPI);
+    event.eventDuration3 = WebPageLoadTimesCollector.getLoadEventStart(this.currentPerformanceAPI, this.oldPerformanceAPI);
+    event.eventDuration4 = WebPageLoadTimesCollector.getDOMComplete(this.currentPerformanceAPI, this.oldPerformanceAPI);
+    event.eventDuration5 = WebPageLoadTimesCollector.getLoadEventEnd(this.currentPerformanceAPI, this.oldPerformanceAPI);
+    return event;
+  }
+
+
+  /**
+   *
+   * @param {Object} currentPerformanceAPI
+   * @param {Object} oldPerformanceAPI
+   * @returns {null|Number}
+   */
+  static getDOMInteractive(currentPerformanceAPI, oldPerformanceAPI) {
+    try {
+      if (currentPerformanceAPI) {
+        return Math.round(currentPerformanceAPI.domInteractive);
+      } if (oldPerformanceAPI && oldPerformanceAPI.timing) {
+        /** We need to subtract due to the fact that is in EPOCH Time */
+        return oldPerformanceAPI.timing.domInteractive - oldPerformanceAPI.timing.navigationStart;
+      }
+      return null;
+
+    }
+    catch (e) {
+      // Failed to identify data layer
+      return null;
+    }
+  }
+
+  /**
+   *
+   * @param {Object} currentPerformanceAPI
+   * @param {Object} oldPerformanceAPI
+   * @returns {null|Number}
+   */
+  static getLoadEventStart(currentPerformanceAPI, oldPerformanceAPI) {
+    try {
+      if (currentPerformanceAPI) {
+        return Math.round(currentPerformanceAPI.loadEventStart);
+      } if (oldPerformanceAPI && oldPerformanceAPI.timing) {
+        /** We need to subtract due to the fact that is in EPOCH Time */
+        return oldPerformanceAPI.timing.loadEventStart - oldPerformanceAPI.timing.navigationStart;
```

Comment:
@robertdumitrescu BUG - I may be getting tired... but won't this result in a negative number?

Reply:
the perf timing looks something like this:

`"{"connectStart":1602004610712,"navigationStart":1602004610706,"loadEventEnd":1602004614269,"domLoading":1602004612963,"secureConnectionStart":0,"fetchStart":1602004610712,"domContentLoadedEventStart":1602004613639,"responseStart":1602004612946,"responseEnd":1602004613410,"domInteractive":1602004613638,"domainLookupEnd":1602004610712,"redirectStart":0,"requestStart":1602004610751,"unloadEventEnd":0,"unloadEventStart":0,"domComplete":1602004614260,"domainLookupStart":1602004610712,"loadEventStart":1602004614261,"domContentLoadedEventEnd":1602004613640,"redirectEnd":0,"connectEnd":1602004610712}"`

so nav timing is basically the first metric that is happening. anything else happens after it.